### ISSUE TITLE
Optional `onInit` is mandatory

### ISSUE THREAD

**Issue by dan-lee** [web-ridge/react-ridge-state#8]
Hey! This is a great library and exactly what I was missing for a long time.

Currently I am implementing the persistence layer and I get following TypeScript complaint when trying to implement `onSet`:

> Error:(37, 70) TS2345: Argument of type '{ onSet: (newState: Items[]) => Promise<void>; }' is not assignable to parameter of type 'Options<Items[]>'.
  Property 'onInit' is missing in type '{ onSet: (newState: Items[]) => Promise<void>; }' but required in type 'Options<Items[]>'.

```ts
export const myState = newRidgeState<Items[]>([], {
  onSet: (newState) =>
    AsyncStorage.setItem(PERSISTENCE_KEY, JSON.stringify(newState)),
})
```

When adding `onInit: () => undefined` to options everything is fine.

I guess `onInit` should be an optional argument too.

**Comment by RichardLindhout** [comment]
Oops, that should not be there. I will remove it since it does nothing!

**Comment by RichardLindhout** [comment]
Fixed in 4.1.5 and thanks for your positive feedback!
